src/main/java/org/wildfly/security/hashicorp/vault/VaultConnector.java

```diff
@@ -48,16 +58,35 @@ public void configure() throws VaultException {
 
             this.vault = Vault.create(config);
 
-            if (token == null || token.trim().isEmpty()) {
-                logger.debug("Token is null or empty, using cert auth method");
-                final AuthResponse authResponse = this.vault.auth().loginByCert();
-                if (authResponse == null) {
-                    throw new VaultException("Authentication using client certificates failed! Token was not provided.");
+            AuthResponse authResponse = null;
+
+            try {
+                authResponse = this.vault.auth().loginByCert();
```

Comment:
@honza-kasik  WCan we check if SSL info was provided  before invoking cert auth?  For example `if (sslConfig.isVerify())` or even better would be `if (sslConfig.getSslContext() != null)` - but here it is possible that it is never null, please check

Reply:
I wanted to check what the SSLContext should be and interestingly, I do not see the call `sslConfig.build() ` anywhere, I thought it is required. The vault java library maybe calls it somewhere I can't find now